### ISSUE TITLE
RabbitMQ heartbeat timeout

### ISSUE THREAD

**Issue by jeffadair** [Senzing/stream-loader#181]
**Describe the bug**
We are running into heartbeat timeouts on some rabbitmq connections. Heartbeats are sent on idle connections automatically so this means we are probably hanging onto a thread and never letting pika/rabbitMQ send the heartbeat.

The recommended pattern from the pika maintainer is described here: https://github.com/pika/pika/issues/1104#issuecomment-408680616

Investigate if this is really happening (a quick inspection of the code it looks like we do) and then reorganize the code to run as described in the recommended pattern.

**Comment by jeffadair** [comment]
I've investigated this a bit and here is what I've found. The simple rabbitMQ loader (without info) is fine. It will not have heartbeat problems as it has a single connection that control is passed to (with start_consuming()) and a callback is called when a message arrives. This allows pika to send heartbeat frames if needed.

The withInfo version is vulnerable to heartbeat disconnects. stream-loader [creates three connections](https://github.com/Senzing/stream-loader/blob/32c312cf0fa148ee078f07d812be650cf02ad3c6/stream-loader.py#L1913-L1970): incoming records, outgoing info, outgoing failures. After the three connections are created, we pass control to the incoming records connection like described above. The record connection has control of the thread, but the other two connections get ignored and never get any time to run. In normal operation, we get to adding records and publishing info quickly enough that the info connection stays alive (any activity on a thread is considered a heartbeat, so producing will keep the connection open), but I suspect the failure connection has been dying and we haven't noticed since failures are rare.

I tested this with a test script, and using a single connection with multiple channels instead of three connections, then passing control to the record channel keeps it's associated connection alive.

We currently allow users to specify a different server for each of the three connections. Is this needed? If not, then using one connection for all three channels MIGHT be a solution.

I say 'might' because rabbitMQ recommends using different connections for incoming and outgoing channels so that it can tell publishers to slow down ("back pressure") if the server is overwhelmed ([see recommendation #2 from this article](https://www.cloudamqp.com/blog/2019-11-13-the-relationship-between-connections-and-channels-in-rabbitmq.html)). The way our messages flow might avoid this problem however. An info/failure message is only produced as a result of adding a record. If the server is getting busy, it would send messages to the consumer more slowly, which means records would be added and info messages produced more slowly. If we want to continue allowing for different servers for records, info, and failures, and think this might not be be enough to keep things flowing then we would need a more complex solution (consumer and producer threads with a queue between?)

Other thoughts as I researched this
- We could use one connection for both the info and failure channel. This would reduce our open channels by 1/3 for this use case. 
- RabbitMQ recommends a connection per process (per direction) and a channel per thread. Pika (the python rabbitMQ module) is not threadsafe and recommends one connection per thread. If we wanted to move to a threadsafe module, AMQPStorm seems to be the best maintained, but Rabbitpy exists too. Pika does seem to be the most popular.
